kasa/klaptransport.py

```diff
@@ -355,6 +356,7 @@ async def send(self, request: str):
 
             if TYPE_CHECKING:
                 assert self._encryption_session
+                assert isinstance(response_data, bytes)
```

Comment:
Perhaps httpclient's `post()` should return bytes always, empty `b""` when none is available?